File: jar-common/src/main/java/com/beancounter/common/model/FxRate.java
package com.beancounter.common.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import java.math.BigDecimal;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.RequiredArgsConstructor;

@Data
@Builder
@RequiredArgsConstructor
@AllArgsConstructor
public class FxRate {
  @JsonIgnore
  public static final FxRate ONE = FxRate.builder().rate(BigDecimal.ONE).build();
  private Currency to;
  private Currency from;
  private BigDecimal rate;
  private String date;

}

File: jar-common/src/main/java/com/beancounter/common/model/FxRate.kt
package com.beancounter.common.model

import org.springframework.boot.context.properties.ConstructorBinding
import java.math.BigDecimal

data class FxRate @ConstructorBinding constructor(
        val from: Currency,
        val to: Currency,
        val rate: BigDecimal?,
        val date: String?
)